e2e/tests/snapp_tests.rs

```diff
@@ -266,3 +255,192 @@ fn snapp_auction() {
     );
     println!("Expected trade settlement applied!")
 }
+
+#[test]
+fn snapp_standing_order() {
+    let (eloop, http) = Http::new("http://localhost:8545").expect("transport failed");
+    eloop.into_remote();
+    let web3 = Web3::new(http);
+    let (instance, accounts, _tokens, db) = setup_snapp(&web3, 3, 2, 300);
+
+    let deposit_amount = 300_000_000_000_000_000_000u128;
+    let one_eth = U128::from(1_000_000_000_000_000_000u128);
+
+    let initial_state_hash = instance
+        .get_current_state_root()
+        .wait_and_expect("Could not recover initial state hash");
+
+    println!("Depositing balances for test");
+    instance
+        .deposit(2, U128::from(deposit_amount))
+        .from(Account::Local(accounts[0], None))
+        .wait_and_expect("Failed to send first deposit");
+    instance
+        .deposit(1, U128::from(deposit_amount))
+        .from(Account::Local(accounts[1], None))
+        .wait_and_expect("Failed to send second deposit");
+    wait_for(&web3, 181);
+    let post_deposit_state = await_state_transition(&instance, &initial_state_hash);
+
+    println!("Placing (matching) sell and standing sell orders");
+    instance
+        .place_sell_order(2, 1, one_eth, one_eth)
+        .from(Account::Local(accounts[1], None))
+        .wait_and_expect("Could not place sell order");
+
+    let standing_order_bytes: Vec<u8> = vec![
+        0, 0, 0, 0, 13, 224, 182, 179, 167, 100, 0, 0, // buyAmount=1e18
+        0, 0, 0, 0, 13, 224, 182, 179, 167, 100, 0, 0, // sellAmount=1e18
+        2, 1, // sellToken, buyToken
+    ];
+    instance
+        .place_standing_sell_order(standing_order_bytes)
+        .from(Account::Local(accounts[0], None))
+        .wait_and_expect("Could not place standing order");
+
+    println!("Ensure standing order recorded in DB");
+    wait_for_condition(|| {
+        !db.get_standing_orders_of_slot(&U256::zero()).unwrap()[0]
+            .get_orders()
+            .is_empty()
+    })
+    .expect("Couldn't recover standing order from DB");
+    let standing_orders = db.get_standing_orders_of_slot(&U256::zero()).unwrap();
+    let orders = standing_orders[0].get_orders();
+    println!("Standing Order:\n{:#?}", orders[0]);
+
+    assert_eq!(orders[0].sell_amount, 1_000_000_000_000_000_000u128);
+    assert_eq!(orders[0].buy_amount, 1_000_000_000_000_000_000u128);
+    assert_eq!(orders[0].buy_token, 1);
+    assert_eq!(orders[0].sell_token, 2);
+
+    wait_for(&web3, 181);
+    await_and_fetch_auction_bid(&instance, U256::zero());
+    wait_for(&web3, 181);
+    let post_auction_state = await_state_transition(&instance, &post_deposit_state);
+    let post_auction_state_hash = H256::from_slice(&post_auction_state);
+    println!("Auction 0 settled - {:?}", post_auction_state_hash);
+
+    println!("Ensure standing order account traded");
+    // TODO - let state = await_and_fetch_new_account_state(&db, &post_auction_state_hash);
```

Comment:
That sounds like a nice helper. Can we also improve the error message in the case where we don't observe the state update telling us the last seen state (helps debugging to now where in the test we failed).

Reply:
Yes, I assume you mean better error handling with personalized messages. as opposed to generic fixed ones. I am excited to make this change!

As for your first point, that I should implement `await_and_fetch_new_account_state`, I would have already done this but struggled with passing of the DB into the helper method. Clippy told me at first that it needed to be a boxed reference, but then wouldn't let me. Any suggestions you might have regarding the DB as a parameter of this function would be helpful.

Reply:
@fleupold 

I had already tried to make this helper function a reality, but am still running into some issues that I could use some help with:

Without Boxed:

```
error[E0277]: the size for values of type `(dyn dfusion_core::database::DbInterface + 'static)` cannot be known at compilation time
  --> e2e/src/snapp.rs:91:5
   |
91 |     db: dyn DbInterface,
   |     ^^ doesn't have a size known at compile-time
```

As a reference:

```
error[E0277]: the trait bound `std::boxed::Box<dyn dfusion_core::database::DbInterface>: dfusion_core::database::DbInterface` is not satisfied
   --> e2e/tests/snapp_tests.rs:328:51
    |
328 |     let state = await_and_fetch_new_account_state(&db, post_auction_state_hash);
    |                                                   ^^^ the trait `dfusion_core::database::DbInterface` is not implemented for `std::boxed::Box<dyn dfusion_core::database::DbInterface>`
    |
    = note: required for the cast to the object type `dyn dfusion_core::database::DbInterface`
```

It seems I am not able to pass around Boxes...

The only thing else I can think of is to move the db initialization to the beginning of each test and pass it to the helper as a box...